cluster-state-service/internal/features/wrappers/utils.go

```diff
@@ -40,3 +67,39 @@ func getECSEndpoint() (string, error) {
 
 	return endpoint, nil
 }
+
+func GetLatestECSOptimizedAMIID() (string, error) {
+	region := os.Getenv(regionEnvVarName)
+	if region == "" {
+		return "", fmt.Errorf("Empty region. Please specify the AWS region.")
+	}
+	return latestAMIIDByRegion[region], nil
+
+}
+
+// ValidateECSContainerInstance validates that an ECS instance was launched
+// within a cluster 'clusterName'
+func ValidateECSContainerInstance(ecsWrapper ECSWrapper, clusterName string) error {
+	// Takes some time for EC2 instance to come up and for the ECS agent to
+	// register the container instance with ECS.
+	// Retry ECS list call once every minute for 10 minutes
```

Comment:
Your comment says `Retry ECS list call once every minute for 10 minutes`, but the sleep below is for 30s. Which one is correct?